### ISSUE TITLE
Allow widget breaks in GridView

### ISSUE THREAD

**Issue by Greatcallie** [flutter/flutter#46223]
I would want to add a native ad widget to a GridView after some rows of content. And there is no easy way of doing that at the moment without bringing a lot of complications to my code.

**Comment by iapicca** [comment]
Hi @Greatcallie 
doesn't the [AndroidView](https://api.flutter.dev/flutter/widgets/AndroidView-class.html) class provide the solution you are looking for?
thank you

**Comment by Greatcallie** [comment]
No I am talking about adding a full length widget as a divider in a
GridView after say three rows

**Comment by iapicca** [comment]
Hi again @Greatcallie 
I apologize, I got confused from the "native"  term in the question,
so you would like to achieve something like this, but in a `GridView`, correct?
https://dartpad.dartlang.org/277fa1b7d951c479991039a5b2bc283a
thank you

**Comment by Greatcallie** [comment]
Not really. It's not a consistent seperator. I want to say, separate with a
widget every 5 rows on a GridView.

**Comment by Greatcallie** [comment]
Let's say you wanted an Advert widget after every 10 rows of content on a GridView, what would you do?

**Comment by Piinks** [comment]
This can be done a couple of ways. Either you can define your own grid layout with a SliverGridDelegate, or you could actually extend the existing ones (SliverGridDelegateWithFixedCrossAxisCount, SliverGridDelegateWithMaxCrossAxisExtent) to just modify the row you would like customized.

GridView takes a grid delegate, or you could use a SliverGrid.

If you are suing a SliverGrid, you could go another route and have multiple grids with another sliver in between:

```
CustomScrollView(
  slivers: [
    SliverGrid() // row 1-4
    SliverToBoxAdapter() // OR another sliver
    SliverGrid() // row 6-9
    // and so on...
  ]
)
```

I think we can close this issue, but if we missed anything let us know and we can take another look. Thank you!
